internal/app/secretless/handlers/mysql/connection_details_test.go

```diff
@@ -0,0 +1,83 @@
+package mysql
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConnectionDetailsCorrectlyParsesConnectionDetails(t *testing.T) {
+	credentials := make(map[string][]byte)
+
+	credentials["host"] = []byte("myhost")
+	credentials["port"] = []byte("1234")
+	credentials["username"] = []byte("myusername")
+	credentials["password"] = []byte("mypassword")
+
+	expectedConnDetails := ConnectionDetails{
+		Host:     "myhost",
+		Port:     1234,
+		Username: "myusername",
+		Password: "mypassword",
+		Options:  make(map[string]string),
+	}
+
+	actualConnDetails, err := NewConnectionDetails(credentials)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedConnDetails, *actualConnDetails)
+}
+
+func TestPortDefaultsTo3306IfNotProvided(t *testing.T) {
+	credentials := make(map[string][]byte)
+
+	credentials["host"] = []byte("myhost")
+	credentials["username"] = []byte("myusername")
+	credentials["password"] = []byte("mypassword")
+
+	expectedConnDetails := ConnectionDetails{
+		Host:     "myhost",
+		Port:     3306,
+		Username: "myusername",
+		Password: "mypassword",
+		Options:  make(map[string]string),
+	}
+
+	actualConnDetails, err := NewConnectionDetails(credentials)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedConnDetails, *actualConnDetails)
+}
+
+func TestConnectionDetailsAddsUnknownCredentialsToOptions(t *testing.T) {
+	credentials := make(map[string][]byte)
+
+	credentials["host"] = []byte("myhost")
+	credentials["port"] = []byte("1234")
+	credentials["foo"] = []byte("5432")
+	credentials["username"] = []byte("myusername")
+	credentials["bar"] = []byte("data")
+	credentials["password"] = []byte("mypassword")
+
+	expectedOptions := map[string]string{
+		"foo": "5432",
+		"bar": "data",
+	}
+
+	actualConnDetails, err := NewConnectionDetails(credentials)
+	assert.Nil(t, err)
+
+	assert.EqualValues(t, expectedOptions, (*actualConnDetails).Options)
```

Comment:
`if` guard

Reply:
Also dealt with by the assert.Nil above it